feat(curriculum): add shortest path algorithm to new Python curriculum (#52386)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/20-upcoming-python/learn-algorithm-design-by-building-the-shortest-path-algorithm/655774d01daeeaa1978b99d5.md

---
id: 655774d01daeeaa1978b99d5
title: Step 14
challengeType: 20
dashedName: step-14
---

# --description--

> At the beginning, all the other nodes in the graph are considered to be at infinite distance from the source node, because the distance has not determined yet.

Reviewer: ```suggestion
At the beginning, all the other nodes in the graph are considered to be at infinite distance from the source node, because the distance has not been determined yet.
```

> Create an `else` statement and assign an infinite value to the node in the `distances` dictionary. For that you can use the `float()` function, that generates a floating point number from a number or a string. Pass the string `inf` to the `float()` function.

Reviewer: ```suggestion
Create an `else` statement and assign an infinite value to the node in the `distances` dictionary. For that, you can use `float()` with `inf` as the parameter that generates an infinite positive float value.
```
